core/src/main/scala/org/http4s/headers/Content-Disposition.scala

```diff
@@ -77,9 +80,28 @@ object `Content-Disposition` {
       ci"Content-Disposition",
       v =>
         new Renderable {
+
           def render(writer: Writer): writer.type = {
+            val renderExtFilename =
+              v.parameters.get("filename").exists(!safeChars.matchesAll(_))
+            val withExtParams =
+              if (renderExtFilename && !v.parameters.contains("filename*"))
+                v.parameters + ("filename*" -> v.parameters("filename"))
+              else v.parameters
+            val withExtParamsSorted =
+              if (withExtParams.contains("filename") && withExtParams.contains("filename*"))
+                TreeMap[String, String]() ++ withExtParams
+              else withExtParams
+
             writer.append(v.dispositionType)
-            v.parameters.foreach(p => writer << "; " << p._1 << "=\"" << p._2 << '"')
+            withExtParamsSorted.foreach {
+              case (k, v) if k == "filename" =>
+                writer << "; " << k << '=' << '"'
+                writer.eligibleOnly(v, keep = safeChars, placeholder = '?') << '"'
+              case (k, v) if k.endsWith("*") =>
+                writer << "; " << k << '=' << "UTF-8''" << Uri.encode(v)
+              case (k, v) => writer << "; " << k << "=\"" << v << '"'
+            }
```

Comment:
This is similar enough to the akka-http implementation that it's probably worth an attribution.  We already mention them in `NOTICES` since we've copied elsewhere from them.